projects/fluid_simulation/sim.h

```diff
@@ -67,13 +67,34 @@ class ProjApp;
 class FluidSimulation
 {
 public:
-    FluidSimulation(ProjApp* app);
-    ProjApp*                     GetApp() const { return mApp; }
-    const SimulationConfig&      GetConfig() const;
+    FluidSimulation(ppx::Application* app, ppx::grfx::DevicePtr device, ppx::uint2 resolution, const SimulationConfig& config)
```

Comment:
IHMO: I don't think it's necessary to store a pointer to the application. The device I can understand.

Reply:
Yeah. I'll be removing application ties next.